src/core/Form/ErrorSummary/ErrorSummary.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { MouseEvent, forwardRef, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { HtmlDiv, HtmlDivWithRef, HtmlDivProps, hLevels } from '../../../reset';
+import { IconErrorFilled } from 'suomifi-icons';
+import { AutoId } from '../../utils/AutoId/AutoId';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import {
+  spacingStyles,
+  separateMarginProps,
+  MarginProps,
+} from '../../theme/utils/spacing';
+import { baseStyles } from './ErrorSummary.baseStyles';
+import { Heading } from '../../Heading/Heading';
+import { Link } from '../../Link';
+
+const baseClassName = 'fi-error-summary';
+const inlineAlertClassNames = {
+  styleWrapper: `${baseClassName}_style-wrapper`,
+  content: `${baseClassName}_content`,
+  heading: `${baseClassName}_heading`,
+  textContentWrapper: `${baseClassName}_text-content-wrapper`,
+  icon: `${baseClassName}_icon`,
+  smallScreen: `${baseClassName}--small-screen`,
+};
+
+interface BasicErrorSummaryItemProps {
+  /** Visible text of the error item */
+  text: string;
+}
+
+type InputReferenceProps =
+  | {
+      /** HTML id of an input element. The error corresponds to this input. Provide either this or `inputRef`. */
+      inputId: string;
```

Comment:
Should the naming a descriptions of these be changed in to something more generic to avoid confusion. Would "field" as a word be a more generic refence to where the ref should point? Now the naming and descriptions seem pretty strict even though the typing isn't

Reply:
I think we want to enforce the primary use case of setting focus to inputs. That's why I wouldn't change the prop naming. 

And the comments are actually for internal use only, due to Styleguidist limitations that particular type is not displayed in the documentation page.

Reply:
All good here then :) Let's keep then as they are 👍